frontend/taipy-gui/src/components/Taipy/Metric.tsx

```diff
@@ -91,6 +92,14 @@ const Metric = (props: MetricProps) => {
                             props.max || 100
                         ]
                     },
+                    steps: props.colorMap ? (() => {
+                        const obj = JSON.parse(props.colorMap);
```

Comment:
For simplicity, this whole function Block calculating steps could be put in another useMemo

Reply:
Updated ✍🏻